dl.go

```diff
@@ -24,13 +25,13 @@ type downloadMetadata struct {
 
 // Download begins a download of a url but doesn't block.
 // Returns a future that when called will block until it can return the path to the file on disk or an error.
-func (m *MagnetTarget) DownloadFuture(url string) func() (url string, path string, err error) {
+func (m *MagnetTarget) DownloadFuture(ctx context.Context, url string) func() (url string, path string, err error) {
```

Comment:
Might want to consider adding a context to the future as well to make it return when cancelled, but not super important.

Reply:
Addressed in https://github.com/gravitational/magnet/pull/2/commits/fa3aca08010b89e4bc5351a34cab37f248ff6fbc

Reply:
This was my bad, the context to the base function should actually cancel the download, not sure what I was thinking.